controllers/config_map.go

```diff
@@ -47,70 +49,149 @@ type ConfigMapReconciler struct {
+	ekscontrolplane := &eks.AWSManagedControlPlane{}
+	clusterInfo := &ClusterInfo{}
+
+	capiCluster := &capi.Cluster{}
+	err := r.Client.Get(ctx, req.NamespacedName, capiCluster)
+
 	if err != nil {
 		logger.Error(err, "failed to get cluster")
 		return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
 	}
 
-	if !cluster.DeletionTimestamp.IsZero() {
+	if !capiCluster.DeletionTimestamp.IsZero() {
 		logger.Info("Reconciling delete")
-		return r.reconcileDelete(ctx, cluster)
+		return r.reconcileDelete(ctx, capiCluster)
 	}
 
-	return r.reconcileNormal(ctx, cluster)
+	if IsEKS(*capiCluster) {
+		err := r.Client.Get(ctx, req.NamespacedName, ekscontrolplane)
+		if err != nil {
+			logger.Error(err, "failed to get cluster")
+			return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+		}
+
+		clusterInfo.Name = ekscontrolplane.Name
+		clusterInfo.Namespace = ekscontrolplane.Namespace
+		clusterInfo.Region = ekscontrolplane.Spec.Region
+		clusterInfo.AWSPartition = getPartition(clusterInfo.Region)
+		clusterInfo.VpcID = ekscontrolplane.Spec.NetworkSpec.VPC.ID
+		clusterInfo.RoleArn, err = r.getRoleArn(ctx, ekscontrolplane.Spec.IdentityRef.Name, ekscontrolplane.Namespace)
+		if err != nil {
+			logger.Error(err, "failed to get cluster role identity")
+			return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+		}
+	} else {
+		err = r.Client.Get(ctx, req.NamespacedName, awsCluster)
+		if err != nil {
+			logger.Error(err, "failed to get cluster")
+			return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+		}
+		clusterInfo.Name = awsCluster.Name
+		clusterInfo.Namespace = awsCluster.Namespace
+		clusterInfo.Region = awsCluster.Spec.Region
+		clusterInfo.AWSPartition = getPartition(clusterInfo.Region)
+		clusterInfo.VpcID = awsCluster.Spec.NetworkSpec.VPC.ID
+		clusterInfo.RoleArn, err = r.getRoleArn(ctx, awsCluster.Spec.IdentityRef.Name, awsCluster.Namespace)
+		if err != nil {
+			logger.Error(err, "failed to get cluster role identity")
+			return ctrl.Result{}, errors.WithStack(client.IgnoreNotFound(err))
+		}
+		if sg, ok := awsCluster.Status.Network.SecurityGroups[capa.SecurityGroupControlPlane]; ok {
+			if clusterInfo.SecurityGroups == nil {
+				clusterInfo.SecurityGroups = &crossplaneConfigValuesAWSClusterSecurityGroups{}
+			}
+			clusterInfo.SecurityGroups.ControlPlane = &crossplaneConfigValuesAWSClusterSecurityGroup{
+				ID: sg.ID,
+			}
+		}
+	}
+
+	return r.reconcileNormal(ctx, clusterInfo)
 }
 
-func (r *ConfigMapReconciler) reconcileNormal(ctx context.Context, cluster *capa.AWSCluster) (ctrl.Result, error) {
-	logger := log.FromContext(ctx)
-	logger.Info("Reconciling")
-	defer logger.Info("Done reconciling")
+func IsEKS(cluster capi.Cluster) bool {
+	return cluster.Spec.ControlPlaneRef != nil &&
+		cluster.Spec.ControlPlaneRef.Kind == "AWSManagedControlPlane" &&
+		cluster.Spec.InfrastructureRef != nil &&
+		cluster.Spec.InfrastructureRef.Kind == "AWSManagedCluster"
+}
```

Comment:
```suggestion
func IsEKS(cluster capi.Cluster) bool {
	return cluster.Spec.ControlPlaneRef != nil &&
		cluster.Spec.ControlPlaneRef.Kind == "AWSManagedControlPlane"
}
```

We're not using `AWSManagedCluster` in this operator, so no need to check that part.